### PR TITLE
Backport of  #4559 (Tables for alg params - QGIS Database)

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): #4429. Backport of  #4559.

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 1]
```diff
@@ -21,72 +21,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+     - Defines the encoding of the output layer
+   * - **Overwrite**
+     - ``OVERWRITE``
+     - [boolean]
+       
+       Default: True
+     - If the specified table exists, setting this option to
+       ``True`` will make sure that it is deleted and a new
+       table will be created before the features are added.
+       If this option is ``False``, the features will be
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]: Actually, I'm wrong here. I came across a discussion in a french forum that made me test the alg and if the table already exists, the import just fails. This is not an "append" fallback option.

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: I installed PostgreSQL / PostGIS to test, and the algorithm fails:
> ERROR: relation "mypoly" already exists
To me, this looks like a bug in the algorithm.  So I will leave it and instead report this as a bug in the algorithm.

Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: I don't know whether it's a bug but from ages, appending a PG layer (using DB Manager eg) is a feature request.

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: I have found a relevant QGIS issue (https://github.com/qgis/QGIS/issues/27039), and have created a pull request (https://github.com/qgis/QGIS/pull/33252) to fix the bug.

Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: Great. Thanks!

**Pull request merged**